Wobble/Graphics/Drawable.cs

```diff
@@ -615,6 +623,55 @@ public void ClearAnimations()
         /// <returns></returns>
         public bool IsHovered() => GraphicsHelper.RectangleContains(ScreenRectangle, MouseManager.CurrentState.Position);
 
+        /// <summary>
+        ///     Removes all previously scheduled updates, and schedules a new one to run in the next frame
+        /// </summary>
+        /// <param name="action"></param>
+        public void ScheduleUpdate(Action action)
+        {
+            lock (ScheduledUpdates)
+            {
+                ScheduledUpdates.Clear();
+                ScheduledUpdates.Add(action);
+            }
+        }
+
+        /// <summary>
+        ///     Schedules a new update to be run in the next frame, but does not remove previously scheduled updates.
+        /// </summary>
+        /// <param name="action"></param>
+        public void AddScheduledUpdate(Action action)
+        {
+            lock (ScheduledUpdates)
+                ScheduledUpdates.Add(action);
+        }
+
+        /// <summary>
+        ///     Removes all
+        /// </summary>
+        public void RemoveScheduledUpdates()
+        {
+            lock (ScheduledUpdates)
+                ScheduledUpdates.Clear();
+        }
+
+        /// <summary>
+        ///     Runs all updates that are scheduled for this drawable during <see cref="Update"/>
+        /// </summary>
+        private void RunScheduledUpdates()
+        {
+            lock (ScheduledUpdates)
+            {
+                if (ScheduledUpdates.Count == 0)
+                    return;
+
+                foreach (var update in ScheduledUpdates)
+                    update.Invoke();
+
+                ScheduledUpdates.Clear();
+            }
```

Comment:
If you want to be able to schedule more in these callbacks, the best solution is probably something along
```cs
List<Action> updates;
lock (ScheduledUpdates)
{
    updates = new List<Action>(ScheduledUpdates);
    ScheduledUpdates.Clear();
}

foreach (...) { ... }
```